pdns/misc.cc

```diff
@@ -845,7 +845,6 @@ void addCMsgSrcAddr(struct msghdr* msgh, void* cmsgbuf, const ComboAddress* sour
     memset(pkt, 0, sizeof(*pkt));
     pkt->ipi6_addr = source->sin6.sin6_addr;
     pkt->ipi6_ifindex = itfIndex;
-    msgh->msg_controllen = cmsg->cmsg_len; // makes valgrind happy and is slightly better style
```

Comment:
Well, yes and I wonder why, but valgrind is wrong too. After thinking about it a little more, this patch fixes a known issue and complies with the man page, so it's the right thing to do.